app/models/katello/purpose_addon.rb

```diff
@@ -0,0 +1,18 @@
+module Katello
+  class PurposeAddon < Katello::Model
+    self.table_name = 'katello_purpose_addons'
+    belongs_to :subscription_facet, :inverse_of => :addons, :class_name => 'Katello::Host::SubscriptionFacet'
+
+    def self.find_or_create_by_addons(subscription_facet_id, addon_name)
+      attributes = { subscription_facet_id: subscription_facet_id, name: addon_name }
+
+      Katello::Util::Support.active_record_retry do
```

Comment:
This is a new one to me, is this because we're potentially importing a lot of data from candlepin putting us at risk of the race condition mentioned in the comment on this util function? Just curious.

Reply:
Let's just say I was inspired by some nearby code ... this isn't necessary